neqo-transport/src/crypto.rs

```diff
@@ -243,6 +245,41 @@ impl Crypto {
         self.streams.discard(space);
         self.states.discard(space)
     }
+
+    pub fn create_resumption_token(
+        &mut self,
+        new_token: &mut NewTokenState,
+        tps: &TransportParameters,
+        events: &ConnectionEvents,
+        rtt: u64,
+    ) {
+        if let Agent::Client(ref mut c) = self.tls {
+            if let Some(ref t) = c.resumption_token() {
+                qtrace!("TLS token {}", hex(t.as_ref()));
+                let mut enc = Encoder::default();
+                enc.encode_varint(rtt);
+                enc.encode_vvec_with(|enc_inner| {
+                    tps.encode(enc_inner);
+                });
+                let token = new_token.take_token();
+                enc.encode_vvec(token.as_ref().map_or(&[], |t| &t[..]));
+                enc.encode(t.as_ref());
+                qinfo!("resumption token {}", hex_snip_middle(&enc[..]));
+                events
+                    .client_resumption_token(ResumptionToken::new(enc.into(), t.expiration_time()));
```

Comment:
See previous comment about minimizing knowledge of events in this module.  This can just pass out the token.